cmd/internal/cli/push.go

```
@@ -32,6 +33,25 @@ var PushCmd = &cobra.Command{
 	Args:                  cobra.ExactArgs(2),
 	PreRun:                sylabsToken,
 	Run: func(cmd *cobra.Command, args []string) {
+
+		// if we can load config and if default endpoint is set, use that
+		// otherwise fall back on regular authtoken and URI behavior
+		e, err := sylabsRemote(remoteConfig)
+		if err == nil {
+			authToken = e.Token
+			if !cmd.Flags().Lookup("library").Changed {
+				uri, err := e.GetServiceURI("library")
+				if err != nil {
+					sylog.Fatalf("Unable to get library service URI: %v", err)
+				}
+				PushLibraryURI = uri
+			}
+		} else if err == scs.ErrNoDefault {
+			sylog.Warningf("No default remote in use, falling back to: %v", PushLibraryURI)
+		} else {
+			sylog.Debugf("Unable to load remote configuration: %v", err)
```

If we're unable to load the remote configuration in some way (other than `ErrNoDefault`), I think we should fail rather than a debug message and continuing. If a user runs into this, they'll typically not be running with `--debug` and might not even realize there's something going on. I think this should apply to all new commands which are using this